### PR TITLE
fix the height of select

### LINKED ISSUE #3726: [BUG] height of <select> is not the same like <input>
The height of <select> is not the same like <input> (theme: Origine)
grafik

**To Reproduce**
1. Go to i/?c=configure&a=archiving (Configure: Archiving)

**Expected behavior**
Same height

**Environment information (please complete the following information):**
 - Browser: Firefox 90
 - FreshRSS version: 1.18.1

### PR DESCRIPTION
**Pull request by math-GH**
Closes #3726

Changes proposed in this pull request:
- same height of `<select>` like `<input>`

grafik

How to test the feature manually:
Go to i/?c=configure&a=archiving (Configure: Archiving)

Pull request checklist:

- [ ] clear commit messages
- [x] code manually tested (with Firefox and Microsoft Edge)
- [ ] unit tests written (optional if too hard)
- [ ] documentation updated

### CODE REVIEW

File: p/themes/Origine/origine.css [hunk 1]
```diff
@@ -45,6 +45,10 @@ input, select, textarea {
 	vertical-align: middle;
 }
 
+select {
+	height: 37px;
```
Comment (Alkarex) [inline p/themes/Origine/origine.css]: I am not a found of `px` unit for main dimensions of text-based elements (borders, minimal/maximal dimensions are fine with `px`).
Could you try to use another unit such as `em` ?

Comment (math-GH) [reply p/themes/Origine/origine.css]: I am with you, but the issue is the different box modells of `<input>` and `<select>`

Input: 1px border + 5px padding + 25px height + 5px padding + 1px border = 37px
Select: height = border + padding + X + padding + border

so the height of select needs a `px` height :/

Comment (Frenzie) [reply p/themes/Origine/origine.css]: Have you tried playing around with box-sizing? (content-box for example)

**Comment by Alkarex** [comment]
Hardcoded `px` leads to accessibility problems, e.g. when the text risks not been readable

image

I suggest another fix:

```patch
+++ b/p/themes/Origine/origine.css
@@ -34,14 +34,13 @@ textarea {
 }
 
 input, select, textarea {
-       padding: 5px;
+       padding: 7px;
        background: #fdfdfd;
        color: #666;
        border: 1px solid #bbb;
        border-radius: 3px;
        box-shadow: 0 2px 2px #eee inset;
        min-height: 25px;
-       line-height: 25px;
        vertical-align: middle;
 }
```

**Comment by math-GH** [comment]
> 
> 
> Hardcoded `px` leads to accessibility problems, e.g. when the text risks not been readable
> 
> image
> 
> I suggest another fix:
> 
> ```diff
> +++ b/p/themes/Origine/origine.css
> @@ -34,14 +34,13 @@ textarea {
>  }
>  
>  input, select, textarea {
> -       padding: 5px;
> +       padding: 7px;
>         background: #fdfdfd;
>         color: #666;
>         border: 1px solid #bbb;
>         border-radius: 3px;
>         box-shadow: 0 2px 2px #eee inset;
>         min-height: 25px;
> -       line-height: 25px;
>         vertical-align: middle;
>  }
> ```

grafik

with `box-sizing:box-content` the border will be fine, but the text sticks to the top inside of the `<select>`
grafik

I have no idea how to fix is better. Will investigate it

**Comment by Frenzie** [comment]
Firefox seems to enforce line-heigth: normal. Since there isn't really any reason to specify anything else, just leave out that         min-height and line-height stuff.

**Comment by Frenzie** [comment]
That was presumably some kind of workaround for IE 8 possibly up to 11.

**Comment by Frenzie** [comment]
https://github.com/FreshRSS/FreshRSS/commit/9916d6f79191b227e2d63f974ce96a485a8f6f90 says outdated WebKit problems, same difference.

**Comment by Alkarex** [comment]
Indeed like @Frenzie says, my patch suggestion above was supposed to remove `min-height: 25px;` as well

**Comment by math-GH** [comment]
That helped a lot.

in Firefox it is fine now (input and select height: 32px)
grafik

in Microsoft edge it is not perfect (input height: 31px, select height: 33px)
grafik

**Review by Frenzie — APPROVED** [review]
lgtm

**Comment by Alkarex** [comment]
Thanks @mathContao
A similar fix for the other themes would be welcome

**Pull request merged**